pkg/cmd/cli/cmd/projects.go

```diff
@@ -0,0 +1,182 @@
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	kubecmdconfig "k8s.io/kubernetes/pkg/kubectl/cmd/config"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	cliconfig "github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/project/api"
+
+	"github.com/spf13/cobra"
+)
+
+type ProjectsOptions struct {
+	Config       clientcmdapi.Config
+	ClientConfig *restclient.Config
+	ClientFn     func() (*client.Client, error)
+	Out          io.Writer
+	PathOptions  *kubecmdconfig.PathOptions
+
+	ProjectOnly  bool
+	DisplayShort bool
+
+	// SkipAccessValidation means that if a specific name is requested, don't bother checking for access to the project
+	SkipAccessValidation bool
+}
+
+const (
+	projectsLong = `
+Display information about the current active project and existing projects on the server.
+
+For advanced configuration, or to manage the contents of your config file, use the 'config'
+command.`
+
+	projectsExample = `  # Display the projects that currently exist
+  %[1]s`
+)
+
+// NewCmdProjects implements the OpenShift cli rollback command
+func NewCmdProjects(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &ProjectsOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "projects",
+		Short:   "Display existing projects",
+		Long:    projectsLong,
+		Example: fmt.Sprintf(projectsExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			options.PathOptions = cliconfig.NewPathOptions(cmd)
+
+			if err := options.Complete(f, args, out); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			if err := options.RunProjects(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+	return cmd
+}
+
+func (o *ProjectsOptions) Complete(f *clientcmd.Factory, args []string, out io.Writer) error {
+	var err error
+
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments should be passed")
+	}
+
+	o.Config, err = f.OpenShiftClientConfig.RawConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientConfig, err = f.OpenShiftClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientFn = func() (*client.Client, error) {
+		client, _, err := f.Clients()
+		return client, err
+	}
+
+	o.Out = out
+
+	return nil
+}
+func (o ProjectsOptions) Validate() error {
+	return nil
+}
+
+// RunProjects lists all projects a user belongs to
+func (o ProjectsOptions) RunProjects() error {
+	config := o.Config
+	clientCfg := o.ClientConfig
+	out := o.Out
+
+	currentContext := config.Contexts[config.CurrentContext]
+	currentProject := currentContext.Namespace
+
+	var currentProjectExists bool = false
+
+	client, err := o.ClientFn()
+	if err != nil {
+		return err
+	}
+
+	if len(currentProject) > 0 {
+		if o.DisplayShort {
```

Comment:
This doesn't seem to be set anywhere. If we want to support the short form, we should have a `-q, --short=` flag like in `oc project` that sets `DisplayShort`. Then, I'd say the behavior of short form in `oc projects` should relate to the main purpose of the command (list projects you have) instead of displaying the current one. So probably

```
$ oc projects --short
awesome
notsoawesome
okone
weirdo
```